django/db/models/options.py

```diff
@@ -355,6 +349,47 @@ def swapped(self):
         return None
 
     @cached_property
+    def managers(self):
+        managers = []
+        bases = (b for b in self.model.mro() if hasattr(b, '_meta'))
+        for depth, base in enumerate(bases):
+            for manager in base._meta.local_managers:
+                manager = copy.copy(manager)
+                manager.model = self.model
+                managers.append((depth, manager.creation_counter, manager))
+
+        return make_immutable_fields_list(
+            "managers",
+            (m[2] for m in sorted(managers)),
+        )
+
+    @cached_property
+    def managers_map(self):
+        return {manager.name: manager for manager in reversed(self.managers)}
+
+    @cached_property
+    def base_manager(self):
+        if self.base_manager_name:
+            return self.managers_map[self.base_manager_name]
```

Comment:
Note to self, this should raise a helpful message in case of `KeyError`. Same for `default_manager` and `related_manager` below.